### PR TITLE
Geirst/logd implement rpc forwarder

### PR DESCRIPTION
**Pull request by geirst**
@toregge please review
@havardpe please review first commit (containing rpc integration)
@bjorncs FYI

### CODE REVIEW

File: logd/src/logd/rpc_forwarder.cpp [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+using vespalib::make_string;
+
+namespace logdemon {
+
+RpcForwarder::RpcForwarder(const vespalib::string &hostname, int rpc_port,
+                           double rpc_timeout_secs, size_t max_messages_per_request)
+    : _connection_spec(make_string("tcp/%s:%d", hostname.c_str(), rpc_port)),
+      _rpc_timeout_secs(rpc_timeout_secs),
+      _max_messages_per_request(max_messages_per_request),
+      _supervisor(),
```
Comment (havardpe) [inline logd/src/logd/rpc_forwarder.cpp]: consider keeping the supervisor external to enable re-connect without having to re-create 'the world'

File: logd/src/logd/rpc_forwarder.cpp [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+        request->SubRef();
+        throw ConnectionException(error_msg);
+    }
+    ProtoConverter::ProtoLogResponse proto_response;
+    if (!decode_log_response(*request, proto_response)) {
+        auto error_msg = make_string("Error during decoding of protobuf response from '%s'", _connection_spec.c_str());
+        request->SubRef();
+        throw DecodeException(error_msg);
+    }
+    request->SubRef();
```
Comment (havardpe) [inline logd/src/logd/rpc_forwarder.cpp]: consider using request subref guard

**Review by toregge — CHANGES_REQUESTED** [review]
Looks mostly ok, but need to add all methods before starting the supervisor, as mentioned by @havardpe

**Comment by geirst** [comment]
@toregge PTAL

**Review by toregge — APPROVED** [review]

**Pull request merged**